pkg/networking/middleware/retry_middleware_test.go

```diff
@@ -0,0 +1,90 @@
+package middleware
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+type failRoundtripper struct {
+	actualCount                  uint
```

Comment:
Wondering: is the backoff library using the given function in a goroutine or is it synchronous? If it's async, probably these should be types from `atomic` package or protected by a mutex to avoid race tests noping out.